frontend/src/features/documents/documentUpload/DocumentUploadForm.tsx

```diff
@@ -54,15 +57,24 @@ const DocumentUploadForm: React.FunctionComponent<IDocumentUploadFormProps> = pr
     props.initialDocumentType,
     props.mayanMetadataTypes,
   );
+  useEffect(() => {
+    const isTypeDirty =
```

Comment:
hmm, so it seems like your changes handle the cases where:
the file is changed
the document type is changed

What about metadata changing? or does that already work without explicit handling?

Reply:
it does that automatically given those fields live in formik natively unlike the file. The reason the type needs to be set manually is because it comes as part of the form props